server/graphql/v2/mutation/CreateCollectiveMutation.js

```diff
@@ -22,96 +24,125 @@ async function createCollective(_, args, req) {
-      user: {
-        email: remoteUser.email,
-        collective: remoteUserCollective.info,
-      },
-    },
-  });
+      // Handle GitHub automated approval and apply to the Open Source Collective Host
+      if (args.automateApprovalWithGithub && args.collective.githubHandle) {
+        const githubHandle = args.collective.githubHandle;
+        const opensourceHost = defaultHostCollective('opensource');
+        host = await loaders.Collective.byId.load(opensourceHost.CollectiveId);
+        try {
+          const githubAccount = await models.ConnectedAccount.findOne(
+            {
+              where: { CollectiveId: user.CollectiveId, service: 'github' },
+            },
+            { transaction },
+          );
+          if (!githubAccount) {
+            throw new Error('You must have a connected GitHub Account to create a collective with GitHub.');
+          }
+          // In e2e/CI environment, checkGithubAdmin and checkGithubStars will be stubbed
+          await github.checkGithubAdmin(githubHandle, githubAccount.token);
+          await github.checkGithubStars(githubHandle, githubAccount.token);
+          shouldAutomaticallyApprove = true;
+        } catch (error) {
+          throw new ValidationFailed(error.message);
+        }
+        if (githubHandle.includes('/')) {
+          collectiveData.settings.githubRepo = githubHandle;
+        } else {
+          collectiveData.settings.githubOrg = githubHandle;
+        }
+        collectiveData.tags = collectiveData.tags || [];
+        if (!collectiveData.tags.includes('open source')) {
+          collectiveData.tags.push('open source');
+        }
+      } else if (args.host) {
+        if (!host) {
+          throw new ValidationFailed('Host Not Found');
+        }
+        if (!host.isHostAccount) {
+          throw new ValidationFailed('Host account is not activated as Host.');
+        }
+      }
+      const collective = await models.Collective.create(collectiveData, { transaction });
+      // Add authenticated user as an admin
+      await collective.addUserWithRole(user, roles.ADMIN, { CreatedByUserId: user.id }, {}, transaction);
+      return collective;
+    })
+    .then(async collective => {
+      // Add the host if any
+      if (host) {
+        await collective.addHost(host, user, {
+          shouldAutomaticallyApprove,
+          message: args.message,
+          applicationData: args.applicationData,
+        });
+        purgeCacheForCollective(host.slug);
+      }
 
-  return collective;
+      // Will send an email to the authenticated user OR newly created user (For OCF application)
+      // - tell them that their collective was successfully created
+      // - tell them that their collective is pending validation (which might be wrong if it was automatically approved)
+      const remoteUserCollective = remoteUser
+        ? await loaders.Collective.byId.load(user.CollectiveId)
+        : await models.Collective.findByPk(user.CollectiveId);
+
```

Comment:
I don't get this, `loaders.Collective.byId.load` and `models.Collective.findByPk` are supposed to have the same effect

Reply:
@Betree Oh yeah, thanks for picking up on that. When I was experimenting with the transaction method, `loaders.Collective.byId.load`  was failing for new users because it was within the transaction process, I used `models.Collective.findByPk` to handle the case. But I forgot to remove it after realizing the method call shouldn't be used inside a transaction.